src/models/base.es6.js

```diff
@@ -62,62 +44,50 @@ class Base {
       return true;
     }
 
-    this.emit('validationError', invalid);
     return invalid;
   }
 
-  format (prop, value){
-    if (!this.formatters || !this.formatters[prop]) {
-      return value;
-    }
-
-    return this.formatters[prop](value);
-  }
-
-  toJSON (formatter) {
-    if (formatter && typeof formatter == 'function') {
-      return formatter(this.props);
-    }
+  toJSON () {
+    const type = this._type;
 
-    return this.props;
+    return Object.assign({}, this.props, {
+      _type: type,
+    });
   }
-}
-
-Base.validators = {
-  integer: function(i) {
-    return i === parseInt(i);
-  },
 
-  string: function(s) {
-    return s === s.toString();
-  },
+  static validators = {
+    integer: function(i) {
+      return i === parseInt(i);
+    },
 
-  min: function (i, min) {
-    return i >= min;
-  },
+    string: function(s) {
+      return s === s.toString();
+    },
 
-  max: function (i, max) {
-    return i <= max;
-  },
+    min: function (i, min) {
+      return i >= min;
+    },
 
-  maxLength: function (s, l) {
-    return Base.validators.string(s) && Base.validators.max(s.length, l);
-  },
+    max: function (i, max) {
+      return i <= max;
+    },
 
-  minLength: function (s, l) {
-    return Base.validators.string(s) && Base.validators.min(s.length, l);
-  },
+    maxLength: function (s, l) {
+      return Base.validators.string(s) && Base.validators.max(s.length, l);
+    },
 
-  regex: function(s, expr) {
-    return expr.test(s);
-  },
+    minLength: function (s, l) {
+      return Base.validators.string(s) && Base.validators.min(s.length, l);
```

Comment:
For this and `maxLength`, how would you feel about changing the string check to one for `length` being defined. Then we could use it for arrays as well